src/Cake.Core/IO/IDirectory.cs

```diff
@@ -32,7 +32,8 @@ public interface IDirectory : IFileSystemInfo
         /// Deletes the directory.
         /// </summary>
         /// <param name="recursive">Will perform a recursive delete if set to <c>true</c>.</param>
-        void Delete(bool recursive);
+        /// <param name="force">Will delete read-only files if set to <c>true</c></param>
+        void Delete(bool recursive, bool force);
```

Comment:
@devlead @mholo65 It could be done in internal `DirectoryDeleter` if `IFile` provided access to file attributes but it doesn't. But if exposing getting/setting attributes in `IFile` deserves a breaking change on its own, that could be an alternative way to go.

Please, advise